pkg/reconciler/sequence/resources/eventpolicy.go

```diff
@@ -0,0 +1,117 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	eventingduckv1 "knative.dev/eventing/pkg/apis/duck/v1"
+	eventingv1alpha1 "knative.dev/eventing/pkg/apis/eventing/v1alpha1"
+	flowsv1 "knative.dev/eventing/pkg/apis/flows/v1"
+	messagingv1 "knative.dev/eventing/pkg/apis/messaging/v1"
+	"knative.dev/pkg/kmeta"
+)
+
+const (
+	SequenceChannelEventPolicyLabelPrefix = "flows.knative.dev/"
+	sequenceKind                          = "Sequence"
+	subscriptionKind                      = "Subscription"
+)
+
+func MakeEventPolicyForSequenceChannel(s *flowsv1.Sequence, channel *eventingduckv1.Channelable, subscription *messagingv1.Subscription) *eventingv1alpha1.EventPolicy {
+	return &eventingv1alpha1.EventPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: channel.Namespace,
+			Name:      SequenceEventPolicyName(s.Name, channel.Name),
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: flowsv1.SchemeGroupVersion.String(),
+					Kind:       sequenceKind,
+					Name:       s.Name,
+					UID:        s.UID,
+				},
+			},
+			Labels: LabelsForSequenceChannelsEventPolicy(s.Name),
+		},
+		Spec: eventingv1alpha1.EventPolicySpec{
+			To: []eventingv1alpha1.EventPolicySpecTo{
+				{
+					Ref: &eventingv1alpha1.EventPolicyToReference{
+						APIVersion: channel.APIVersion,
+						Kind:       channel.Kind,
+						Name:       channel.Name,
+					},
+				},
+			},
+			From: []eventingv1alpha1.EventPolicySpecFrom{
+				{
+					Ref: &eventingv1alpha1.EventPolicyFromReference{
+						APIVersion: messagingv1.SchemeGroupVersion.String(),
+						Kind:       subscriptionKind,
+						Name:       subscription.Name,
+						Namespace:  subscription.Namespace,
+					},
+				},
+			},
+		},
+	}
+}
+
+func LabelsForSequenceChannelsEventPolicy(sequenceName string) map[string]string {
+	return map[string]string{
+		SequenceChannelEventPolicyLabelPrefix + "sequence-name": sequenceName,
+	}
+}
+
+func SequenceEventPolicyName(sequenceName, postfix string) string {
+
+	if postfix == "" {
+		return sequenceName
+	}
+	return kmeta.ChildName(sequenceName, "-"+postfix)
+
+}
+
+// MakeEventPolicyForSequenceInputChannel creates an EventPolicy for the input channel of a Sequence
+func MakeEventPolicyForSequenceInputChannel(s *flowsv1.Sequence, inputChannel *eventingduckv1.Channelable, sequencePolicy *eventingv1alpha1.EventPolicy) *eventingv1alpha1.EventPolicy {
+	return &eventingv1alpha1.EventPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: inputChannel.Namespace,
+			Name:      SequenceEventPolicyName(s.Name, sequencePolicy.Name),
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: flowsv1.SchemeGroupVersion.String(),
+					Kind:       sequenceKind,
+					Name:       s.Name,
+					UID:        s.UID,
```

Comment:
another option here, is to set the owner reference to the policy which applies to the sequence (`sequencePolicy`). This has the advantage, that this "new" policy gets automatically cleaned up, when `sequencePolicy` gets deleted by a user...

Reply:
Currently this feature should be working as the "new' policy should be cleaned up [here](https://github.com/knative/eventing/pull/8106/files/9d1e73b40cb44de69c8681706f6330b0ef7b3307#diff-643da9b273b44790c71a3ed7a606f802c21831439d9bc745407eb63191a3e8f8R393).

But change that OwnerReference to the Sequence's policy makes more sense, as they are connected.